.github/workflows/release.yaml

```diff
@@ -0,0 +1,32 @@
+name: Release Partner Charts
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Configure git
+      run: |
+        git config --global user.name "github-actions[bot]"
+        git config --global user.email "github-actions[bot]@users.noreply.github.com"
+
+    - name: Update main-source branch
+      run: |
+        scripts/pull-ci-scripts
+        bin/partner-charts-ci auto
+        git push origin main-source
```

Comment:
Instead of pushing directly to the branches, wouldn't it be better to create PRs for each individual change, with auto-merge enabled?

Reply:
What do you mean by "each individual change"? Are you suggesting that this automation should create a PR that is merged automatically every time it runs? If so, I can't think of a benefit to doing this - can you please elaborate?

If this isn't what you meant, can you clarify?

Reply:
By creating a PR (even with auto merge enabled), it forces an execution of the CI scripts to validate the change. Currently, this executes `partner-charts-ci validate` (see [.github/workflows/pull-request.yml](https://github.com/rancher/partner-charts/blob/main-source/.github/workflows/pull-request.yaml)). If something went wrong, we would have the ability to stop the change from getting merged.